apisix/plugins/cors.lua

```diff
@@ -151,6 +152,17 @@ local function set_cors_headers(conf, ctx)
     end
 end
 
+local function get_match_domain(allow_origins, req_origin)
+    if allow_origins == req_origin or allow_origins == '*' then
+        return req_origin
+    else
+        local matched = re_find(req_origin, allow_origins, "jo")
```

Comment:
We need an extra configuration called `allow_origins_by_regex`. And this configuration should be an array. Please take a look at plugin uri-blocker.

Most importantly, please add tests and docs. We can't accept a new feature without them.

Reply:
ok , but when user set  `add allow_origins_by_regex` and `allow_origins` at the same time, how can we deal with that. they may have conflicts

Reply:
Try `allow_origins` first, if mismatched, try `allow_origins_by_regex`.

Reply:
ok